internal/pkg/deploy/pipeline.go

```diff
@@ -72,14 +72,23 @@ func (a *ArtifactBucket) Region() (string, error) {
 	return parsedArn.Region, nil
 }
 
-// GitHubSource defines the (GH) source of the artifacts to be built and deployed.
 type GitHubSource struct {
+	ProviderName  string
+	Branch        string
+	RepositoryURL GitHubURL
+}
+
+// GitHubSource defines the (GH) source of the artifacts to be built and deployed.
+type GitHubV1Source struct {
 	ProviderName                string
 	Branch                      string
-	RepositoryURL               string
+	RepositoryURL               GitHubURL
 	PersonalAccessTokenSecretID string
 }
 
+// GitHubURL is the common type for repo URLs for both GitHubSource versions
```

Comment:
nit: 
```suggestion
// GitHubURL is the common type for repo URLs for both GitHubSource versions.
```
https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences